rstar/src/geopoint.rs

```diff
@@ -0,0 +1,112 @@
+use crate::PointExt;
+use geo_types;
+
+type LonLatFloat = f32;
+
+#[derive(PartialEq, Clone, Copy, Debug, Default)]
+struct GeoPoint(geo_types::Point<LonLatFloat>);
+
+impl GeoPoint {
+    const RE: f64 = 6378.137; // equatorial radius in km
+    const FE: f64 = 1.0 / 298.257223563; // flattening
+    const E2: f64 = Self::FE * (2.0 - Self::FE);
+
+    pub fn new(x: LonLatFloat, y: LonLatFloat) -> Self {
+        GeoPoint(geo_types::Point::new(x, y))
+    }
+
+    fn x(&self) -> LonLatFloat {
+        self.0.x()
+    }
+
+    fn y(&self) -> LonLatFloat {
+        self.0.y()
+    }
+
+    fn long_diff(a: LonLatFloat, b: LonLatFloat) -> LonLatFloat {
+        let threesixty = 360.;
+        let diff = a - b;
+        diff - ((diff / threesixty).round() * threesixty)
+    }
+
+    fn multipliers_for_lonlat_converstion(latitude: LonLatFloat) -> (LonLatFloat, LonLatFloat) {
+        let one = 1.;
+        let e2 = Self::E2 as LonLatFloat;
+
+        // Curvature formulas from https://en.wikipedia.org/wiki/Earth_radius#Meridional
+        let coslat = latitude.to_radians().cos();
+        let w2 = one / (one - e2 * (one - coslat * coslat));
+        let w = w2.sqrt();
+
+        // multipliers for converting longitude and latitude degrees into distance
+        let dkx = w * coslat; // based on normal radius of curvature
+        let dky = w * w2 * (one - e2); // based on meridonal radius of curvature
+
+        Self::calculate_multipliers(dkx, dky)
+    }
+
+    fn calculate_multipliers(dkx: LonLatFloat, dky: LonLatFloat) -> (LonLatFloat, LonLatFloat) {
+        let re = Self::RE as LonLatFloat;
+        let mul = (1000. as LonLatFloat).to_radians() * re;
+        let kx = mul * dkx;
+        let ky = mul * dky;
+        (kx, ky)
+    }
+}
+
+impl PointExt for GeoPoint {
+    fn sub(&self, other: &Self) -> Self {
```

Comment:
Isn't this definition inconsistent with the various default implementations defined by the `PointExt` trait?

Reply:
It's just a way how I introduced custom distance implementation for lon/lat coordinates.

Reply:
Thank you for the link to the paper! I'll check it out.